clients/admin-ui/src/features/system/AddMultipleSystems.tsx

```diff
@@ -0,0 +1,249 @@
+              onChange: table.getToggleAllRowsSelectedHandler(),
+              manualDisable: allRowsAdded,
+            }}
+          />
+        ),
+        cell: ({ row }) => (
+          <IndeterminateCheckboxCell
+            {...{
+              checked: row.getIsSelected(),
+              disabled: !row.getCanSelect(),
+              indeterminate: row.getIsSomeSelected(),
+              onChange: row.getToggleSelectedHandler(),
+              initialValue: row.original.linked_system,
+            }}
+          />
+        ),
+        meta: {
+          width: "55px",
+        },
+      }),
+      columnHelper.accessor((row) => row.legal_name, {
+        id: "legal_name",
+        cell: (props) => <DefaultCell value={props.getValue()} />,
+        header: (props) => <DefaultHeaderCell value={systemText} {...props} />,
+      }),
+      columnHelper.accessor((row) => row.vendor_id, {
+        id: "vendor_id",
+        cell: (props) => <VendorSourceCell value={props.getValue()} />,
+        header: (props) => <DefaultHeaderCell value="Source" {...props} />,
+        meta: {
+          width: "80px",
+        },
+      }),
+    ],
+    [allRowsAdded, systemText]
+  );
+
+  const rowSelection = useMemo(() => {
+    const innerRowSelection: Record<string, boolean> = {};
+    dictionaryOptions.forEach((ds, index) => {
+      if (ds.linked_system) {
+        innerRowSelection[index] = true;
+      }
+    });
+    return innerRowSelection;
+  }, [dictionaryOptions]);
+
+  const tableInstance = useReactTable<MultipleSystemTable>({
+    columns,
+    data: dictionaryOptions,
+    getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    getFilteredRowModel: getFilteredRowModel(),
+    getPaginationRowModel: getPaginationRowModel(),
+    onGlobalFilterChange: setGlobalFilter,
+    enableRowSelection: true,
+    enableSorting: true,
+    enableGlobalFilter: true,
+    state: {
+      globalFilter,
+    },
+    initialState: {
+      rowSelection,
+      pagination: {
+        pageSize: 25,
+      },
+    },
+  });
+
+  const addVendors = async () => {
+    const vendorIds = tableInstance
+      .getSelectedRowModel()
+      .rows.filter((r) => !r.original.linked_system)
+      .map((r) => r.original.vendor_id);
+    if (vendorIds.length > 0) {
+      await postVendorIds(vendorIds);
+      router.push(redirectRoute);
+      toast(
+        successToastParams(
+          `Successfully added ${
+            vendorIds.length
+          } ${systemText.toLocaleLowerCase()}`
+        )
+      );
+    }
+  };
+
+  const anyNewSelectedRows = tableInstance
+    .getSelectedRowModel()
+    .rows.some((row) => !row.original.linked_system);
+
+  if (isPostLoading || isPostSuccess) {
+    return (
+      <Flex height="100%" justifyContent="center" alignItems="center">
+        <Spinner />
+      </Flex>
+    );
+  }
+
+  if (isGetLoading) {
+    return <TableSkeletonLoader rowHeight={36} numRows={15} />;
```

Comment:
The skeleton loader is only loaded when the table itself is having its data loaded.

The post route gets a spinner because it redirects to a new page after it finishes.